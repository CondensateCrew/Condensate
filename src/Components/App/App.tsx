import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.scss';
import WelcomePage from '../WelcomePage/WelcomePage';
import Dashboard from '../../Containers/Dashboard/Dashboard';
import RoundOne from 'Containers/RoundOne/RoundOne';
import RoundTwo from 'Containers/RoundTwo/RoundTwo';
import Instructions from 'Containers/Instructions/Instructions';

const App: React.FC = () => {
  return (
    <div className="App">
      <Switch>
        <Route exact path='/' render={() => <WelcomePage />}/>
        <Route path='/dashboard' render={() => <Dashboard />}/>
        <Route path='/gameboard/round-two' render={() => <RoundTwo />}/>
        <Route path='/instructions' render={() => <Instructions />}/>
        <Route path='/round-one' render={() => <RoundOne />}/>
        <Route component={() => <div>Not Found</div>} />
      </Switch>
    </div>
  );
}

export default App;
