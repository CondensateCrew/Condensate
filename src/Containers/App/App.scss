@import "../../css/variables.scss";

.App {
  text-align: center;
  height: 100vh;
  &>main {
    padding: $medium-space;
    height: 100%;
  }
}
