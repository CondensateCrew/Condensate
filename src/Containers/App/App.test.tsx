import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

let mockUser: Object = {};
jest.mock('react-redux', () => ({
  useSelector: () => mockUser
}));

describe('App component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if there is user logged', () => {
    mockUser = {
      id: '12dqdw2',
      firstName: 'Ray',
      lastName: 'Z'
    };
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
