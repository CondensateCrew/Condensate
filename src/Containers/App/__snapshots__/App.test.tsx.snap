// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match the snapshot 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Redirect
      to="/"
    />
    <Route
      path="/dashboard"
      render={[Function]}
    />
    <Route
      path="/instructions/:type"
      render={[Function]}
    />
    <Route
      path="/round-one"
      render={[Function]}
    />
    <Route
      path="/round-two"
      render={[Function]}
    />
    <Route
      path="/round-three"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;

exports[`App component should match the snapshot if there is user logged 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/dashboard"
      render={[Function]}
    />
    <Route
      path="/instructions/:type"
      render={[Function]}
    />
    <Route
      path="/round-one"
      render={[Function]}
    />
    <Route
      path="/round-two"
      render={[Function]}
    />
    <Route
      path="/round-three"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
